### PR TITLE
TSPS-220 get Teaspoons into backstage

### PR DESCRIPTION
**Pull request by mmorgantaylor**
### Description 

dsp is starting to use [backstage](https://backstage.dsp-devops.broadinstitute.org/catalog?filters%5Bkind%5D=component) to help with, among other things, tracking the production readiness of a repo.  We want to add our repo to backstage which involves adding a foundation.yml file to our repo.

to test this out, I've pointed backstage at the foundation.yaml file in this branch; the resulting component is [here](https://backstage.dsp-devops.broadinstitute.org/catalog/default/component/teaspoons). once this PR is merged, we'll probably want to unregister that one and point backstage to our base repo (rather than a specific branch), at which point backstage will find that file automatically.

### Jira Ticket
https://broadworkbench.atlassian.net/browse/TSPS-220

### CODE REVIEW

File: foundation.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  annotations:
+    github.com/project-slug: databiosphere/terra-scientific-pipelines-service
+  links:
+    - url: https://grafana.dsp-devops.broadinstitute.org/d/3VhQ1_0Sz/teaspoons?orgId=1
+      title: Teaspoons Grafana Dashboard
+      icon: dashboard
+spec:
+  type: service
+  lifecycle: experimental
+  owner: teaspoons
```
Comment (mmorgantaylor) [inline foundation.yaml:25]: @choover-broad do you know how to create a new group in backstage? or should we just put devops here?

File: foundation.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  annotations:
+    github.com/project-slug: databiosphere/terra-scientific-pipelines-service
+spec:
+  type: openapi
+  lifecycle: experimental
+  system: terra
+  owner: teaspoons
+  definition:
+    $text: ./common/openapi.yml
+---
```
Comment (jsotobroad) [inline foundation.yaml:53]: newline :)

File: foundation.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      title: Teaspoons Grafana Dashboard
+      icon: dashboard
+spec:
+  type: service
+  lifecycle: experimental
+  owner: teaspoons
+  system: terra
+  dependsOn:
+    - component:sam
+    - component:terra-workspace-manager
```
Comment (jsotobroad) [inline foundation.yaml:29]: i think we depend on more currently

Comment (mmorgantaylor) [reply foundation.yaml:29]: yeah, though wsm is the only other one that's actually in backstage right now. (sam was in the bmp template so i kept it but there's not a sam backstage component.) not sure whether to add them all even if they don't exist

Comment (jsotobroad) [reply foundation.yaml:29]: yea if we have sam and they dont have a backstage component then i would add the rest too

**Review by jsotobroad — APPROVED** [review]

**Pull request merged**